xarray/core/duck_array_ops.py

```diff
@@ -351,6 +351,32 @@ def f(values, axis=None, skipna=None, **kwargs):
 _mean = _create_nan_agg_method("mean")
 
 
+def _datetime_nanmin(array):
+    """nanmin() function for datetime64.
+
+    Caveats that this function deals with:
+
+    - In numpy < 1.18, min() on datetime64 incorrectly ignores NaT
+    - numpy nanmin() don't work on datetime64 (all versions at the moment of writing)
+    - dask min() does not work on datetime64 (all versions at the moment of writing)
+    """
+    from .dataarray import DataArray
+    from .variable import Variable
+
+    if isinstance(array, (DataArray, Variable)):
```

Comment:
does this ever happen? All the other functions in this file work at the numpy, dask level (I think) so maybe this shouldn't be necessary.

Reply:
Yes, I was surprised myself. I didn't investigate, but I agree it's not clean.

Reply:
Fixed it